widget/table.go

```diff
@@ -687,7 +706,17 @@ func (t *Table) finishScroll() {
 func (t *Table) hoverAt(pos fyne.Position) {
 	col := t.columnAt(pos)
 	row := t.rowAt(pos)
-	t.hoveredCell = &TableCellID{row, col}
+
+	oldHover := t.hoveredCell
+	var newHover *TableCellID
+	if row >= 0 && col >= 0 {
+		newHover = &TableCellID{row, col}
+	}
+
+	if oldHover != nil && newHover != nil && *oldHover == *newHover {
+		return
```

Comment:
Doesn't this return break hovering of headers?

Reply:
hmm, this basically makes it less spammy without, making sure it only triggers the event if there is effectively a change in the hover, otherwise every time you move the mouse inside a cell it triggers the event. 

I may be wrong, but If hovering over headers was supposed to be working, then it isn't. I've checked with the demo app, hovering over the headers does nothing, just like it does nothing with the current code, same by moving around the table with the direction keys it never goes over the headers. 

Fyne demo app @latest

https://github.com/user-attachments/assets/8265db55-dad0-48e5-a6d0-482a4c05f4db

My little demo/testing app: 


https://github.com/user-attachments/assets/01dda4ff-df78-475c-85db-32de93f25e5b

Reply:
Alright, guess this should do it!, the callback and the check for changes is only done at the end!